common/resource_manager.cc

```diff
@@ -520,4 +531,137 @@ bool ResourceManager::CheckAllowNavigation(const std::string& url) {
   return result = false;
 }
 
+bool ResourceManager::isEncrypted(const std::string& path) {
+  auto setting = application_data_->setting_info();
+  if (setting.get() == NULL)
+    return false;
+
+  if (setting->encryption_enabled()) {
+    std::string ext = utils::ExtName(path);
+    if (kEncryptedFileExtensions.count(ext) > 0) {
+      LOGGER(DEBUG) << "Resource have been encrypted";
+      return true;
+    }
+  }
+  return false;
+}
+
+std::string ResourceManager::DecryptResource(const std::string& path) {
+  int ret;
+  // read file and make a buffer
+    std::string src_path(path);
+  if (utils::StartsWith(src_path, kSchemeTypeFile)) {
+    src_path.erase(0, strlen(kSchemeTypeFile));
+  }
+
+  FILE *src = fopen(src_path.c_str(), "rb");
+  if (!src) {
+    LOGGER(ERROR) << "Cannot open file for decryption: " << src_path;
+    return std::string();
+  }
+
+  // Read filesize
+  fseek(src, 0, SEEK_END);
+  size_t src_len = ftell(src);
+  rewind(src);
+
+  // Read buffer from the source file
+  char* src_buf = new char[src_len];
+  if (src_len != fread(src_buf, sizeof(char), src_len, src)) {
```

Comment:
I have a question. If src_len is a huge number(9999999...) and fread's src_len is the huge number, is it okay? 
I don't know whether it's okay or not if fread() reads data by one time. 
Is it okay to ignore that cotext switching fires while reading the data by fread()? (fread uses locking)